python/lsst/ts/mtaos/mtaos.py

```diff
@@ -1006,6 +1020,44 @@ async def run_closed_loop(self):
                         self.log.info(f"Visit {visit_id} already processed, skipping.")
                         continue
 
+                    filter_label, elevation = await self.model.get_image_info(
+                        visit_id,
+                        self.camera_name,
+                    )
+
+                    rotation_angle = float(
+                        np.mean(np.array(self.image_rotator[image_in_oods.obsid]))
+                    )
+
+                    # Since the current rotator and elevation positions
+                    # are being updated in the background, we need to
+                    # create a local copy and assume it's static
+                    # in the following if statement.
+                    current_rotator_position = self.current_rotator_position
+                    current_elevation_position = self.current_elevation_position
+
+                    if (
+                        (
+                            np.abs(rotation_angle - current_rotator_position)
+                            > self.rotation_angle_limit
+                        )
+                        or (
+                            np.abs(elevation - current_elevation_position)
+                            > self.elevation_angle_limit
+                        )
+                        or self.current_elevation_position is None
```

Comment:
I don't think this is going to work. If `self.current_elevation_position` is `None` this is raise an exception in line 1041 above. I think you want to do something like

```
if current_elevation_position is None or current_rotator_position is None or (...):
    ...
```

This way Python will stop evaluating the expression if either `current_elevation_position` or `current_rotator_position` are `None`.

Also, make sure you use the local variable and not the class attribute.